src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java

```diff
@@ -238,9 +250,29 @@ private List<ErrorInfoDto> addStepErrors(Map<Status, Map<String, List<ErrorLogVi
     }
 
     private static String createMessageRecordError(RecordEntity recordEntity) {
-
         return Stream.of(recordEntity.getEuropeanaId(), recordEntity.getProviderId()).filter(
                 Objects::nonNull).filter(id -> !id.isBlank()).collect(Collectors.joining(" | "));
     }
 
+    private TiersZeroInfo prepareTiersInfo(String datasetId){
+        // get list of records with content tier 0
+        List<String> listOfRecordsIdsWithContentZero = recordRepository.findByDatasetIdAndContentTier(datasetId, MediaTier.T0.toString());
+        // get list of records with metadata tier 0
+        List<String> listOfRecordsIdsWithMetadataZero = recordRepository.findByDatasetIdAndMetadataTier(datasetId, MetadataTier.T0.toString());
```

Comment:
There should be a way to limit the querying to just 10 records, instead of getting all records and then just taking the first 10? Seems like overkill (and a waste of IO).

Reply:
Why I gathered everything first and then get the top 10 was because I wanted to get the count of the total records as well.
But that is a good point, I have created new methods in the record repository to gather the top 10 records as well as counting the number of records